### PR TITLE
feat: tweak keyfunder params

### PR DESCRIPTION
**Pull request by paulbalaji**
resolves https://github.com/hyperlane-xyz/issues/issues/1103

- tune down eth L1 required balance
- tune down L2s required balance
- improve comment r.e. 5x L1->L2 multiplier
- update + tidy up L2Chains types

### CODE REVIEW

File: typescript/infra/config/environments/mainnet3/funding.ts [hunk 1]
```diff
@@ -26,20 +26,21 @@ export const keyFunderConfig: KeyFunderConfig = {
   connectionType: RpcConsensusType.Single,
   // desired balance config
   desiredBalancePerChain: {
-    arbitrum: '3',
     avalanche: '5',
-    base: '3',
     bsc: '5',
     celo: '3',
-    ethereum: '5',
+    ethereum: '0.5',
```
Comment (aroralanuk) [inline typescript/infra/config/environments/mainnet3/funding.ts:32]: there's no boost for l1 iirc?

File: typescript/infra/scripts/funding/fund-keys-from-deployer.ts [hunk 2]
```diff
@@ -539,19 +542,19 @@ class ContextFunder {
   }
 
   private async bridgeIfL2(chain: ChainName) {
-    if (L2Chains.includes(chain)) {
+    if (L2Chains.includes(chain as L2Chain)) {
```
Comment (aroralanuk) [inline typescript/infra/scripts/funding/fund-keys-from-deployer.ts:545]: would prefer not using the "as" cast here

File: typescript/infra/scripts/funding/fund-keys-from-deployer.ts [hunk 3]
```diff
@@ -65,29 +65,29 @@ const nativeBridges = {
...
-  | Chains.optimism
-  | Chains.optimismgoerli
-  | Chains.arbitrum
-  | Chains.arbitrumgoerli
-  | Chains.base;
-
-const L2Chains: ChainName[] = [
-  Chains.optimism,
-  Chains.optimismgoerli,
+type L2Chain = (typeof L2Chains)[number];
```
Comment (aroralanuk) [inline typescript/infra/scripts/funding/fund-keys-from-deployer.ts:68]: it's better to divide the l2 as OpStack, ArbNitro, etc which all have the same funding mechanism (contract interaction) and then union them to get L2Chains

File: typescript/infra/config/environments/mainnet3/funding.ts (hunk 1 above)
Comment (paulbalaji) [reply typescript/infra/config/environments/mainnet3/funding.ts:32]: yup, but lowering this back down due to levels of current use + cost of eth

File: typescript/infra/scripts/funding/fund-keys-from-deployer.ts (hunk 3 above)
Comment (paulbalaji) [reply typescript/infra/scripts/funding/fund-keys-from-deployer.ts:68]: good idea

File: typescript/infra/scripts/funding/fund-keys-from-deployer.ts (hunk 2 above)
Comment (paulbalaji) [reply typescript/infra/scripts/funding/fund-keys-from-deployer.ts:545]: alternate would be to do `(L2Chains as string[])` instead

Comment (paulbalaji) [reply typescript/infra/scripts/funding/fund-keys-from-deployer.ts:545]: also why not cast here, as we're casting a few lines later

Comment (paulbalaji) [reply typescript/infra/scripts/funding/fund-keys-from-deployer.ts]: rejigged stuff but it means we're doing a nasty `as readonly string[]` when switching depending on the L2 chain bridge we need

Comment (aroralanuk) [reply typescript/infra/scripts/funding/fund-keys-from-deployer.ts]: ok, this works then

**Pull request merged**